connector/elasticapmconnector/connector.go

```diff
@@ -0,0 +1,81 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package elasticapmconnector // import "github.com/elastic/opentelemetry-collector-components/connector/elasticapmconnector"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/connector"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/processor"
+
+	"github.com/elastic/opentelemetry-collector-components/connector/signaltometricsconnector"
+	"github.com/elastic/opentelemetry-collector-components/processor/lsmintervalprocessor"
+)
+
+var (
+	lsmintervalFactory     = lsmintervalprocessor.NewFactory()
+	signaltometricsFactory = signaltometricsconnector.NewFactory()
+)
+
+type elasticapmConnector struct {
+	cfg         *Config
+	set         connector.Settings
+	lsminterval processor.Metrics
+}
+
+func newElasticAPMConnector(
+	ctx context.Context,
+	cfg *Config,
+	set connector.Settings,
+	nextConsumer consumer.Metrics,
+) (*elasticapmConnector, error) {
+	c := &elasticapmConnector{cfg: cfg, set: set}
```

Comment:
Nit: Can move this after `lsminterval` is created to make the error case more obvious